crates/task/src/dependency.rs

```diff
@@ -131,12 +136,21 @@ impl<T: Clone + Send + Sync + 'static> EventDependency<T> {
         Self {
             event_rx: receiver,
             match_fn: Box::new(match_fn),
+            completed_dependency: None,
         }
     }
+
+    /// Mark a dependency as completed.
+    pub fn mark_as_completed(&mut self, dependency: T) {
```

Comment:
I _think_ I don't mind this change. We associate this task as `done` right away and provide it with a value to return. This should, in theory, return the proper result when it comes time to collect everyone. This was what I alluded to above, I need a way to, for example, initiate a proposal off of a `SendPayloadCommitmentAndMetadata` or `ViewSyncFinalize...` event, but I _also_ need to be able to capture that data. Sending this through a bespoke init function for each type would be cumbersome at best. So I came up with this which essentially just blasts the exact value back to the vec.